utils.js

```diff
@@ -1,15 +1,69 @@
 const { not, equals, pick } = require('ramda')
 
-async function createTable({ dynamodb, name, attributeDefinitions, keySchema }) {
+const validate = {
+  streamViewType: (inputs) => {
+    if (!inputs.streamViewType) {
+      return
+    }
+
+    const validStreamTypes = ['NEW_IMAGE', 'OLD_IMAGE', 'NEW_AND_OLD_IMAGES', 'KEYS_ONLY']
+    if (!validStreamTypes.includes(inputs.streamViewType)) {
+      throw Error(`${inputs.streamViewType} is not a valid streamViewType.`)
+    }
+  },
+
+  streamViewTypeUpdate: (comp, previousTable, inputs) => {
+    if (!previousTable.streamArn || !inputs.streamViewType) {
+      return
+    }
+
+    if (comp.state.stream && inputs.stream && comp.state.streamViewType !== inputs.streamViewType) {
+      throw Error(`You cannot change the view type of an existing DynamoDB stream.`)
+    }
+  }
+}
+
+async function getStreamArn ({dynamodb, name}) {
+  const maxTries = 5
+  let tries = 0
+
+  const getStreamArn = async () => { 
+    if (tries > maxTries) {
+      throw Error(`There was a problem getting the arn for your DynamoDB stream. Please try again.`)
+    }
+
+    const {streamArn } = await describeTable({ dynamodb, name})
+    if (!streamArn && tries <= maxTries) {
+        tries++
+        const sleep = ms => new Promise(r => setTimeout(r,ms))
+        await sleep(3000)
+        return await getStreamArn()
+    }
+    return streamArn
+  }
+
+  return await getStreamArn()
+}
+
```

Comment:
This function is used in the update table function below. 
I found that if I created a table with a stream, I would get the `data. TableDescription.LatestStreamArn` successfully. But if I created a table with no stream, then added it later in the update, `data. TableDescription.LatestStreamArn` from the `updateTable` function would return undefined (I am assuming that is because it takes a bit of time to register?). Running `sls` a second time will correctly give you the stream arn.

Anyways, that is a problem, because users will likely depend on getting a stream arn back from the outputs of this component. So the above function will try to get the stream arn, if it doesnt, it will wait 3 seconds and try again. It will do this 5 more times until it gets a stream arn. If it does not get it in 5 tries, it will throw an error.

The error doesnt mean it wasnt successfully created, it just means it did not get the stream arn, which might be really important if the user is depending on the stream arn for other components.